src/app/project-admin/component/project-admin.component.ts

```diff
@@ -0,0 +1,129 @@
+    this.project$.subscribe((project) => this.project = project);
+
+    NotificationService.subscribeToNotification(this, {
+      projectId: projectId,
+      whitelist: ["pat"],
+      func: this.handleWebsocketNotification
+    });
+
+    let preparationTasks$ = [];
+    preparationTasks$.push(this.preparePersonalAccessTokensRequest(projectId));
+
+  }
+
+  private checkUserIsAuthorized() {
+    if (!ConfigManager.isInit()) {
+      timer(250).subscribe(() => this.checkUserIsAuthorized());
+      return;
+    }
+
+    if (!ConfigManager.getIsAdmin()) {
+      console.log("you shouldn't be here")
+      this.router.navigate(["projects"])
+    }
+  }
+
+  preparePersonalAccessTokensRequest(projectId: string) {
+    let personalAccessTokens$;
+    [this.personalAccessTokensQuery$, personalAccessTokens$] = this.projectApolloService.getAllPersonalAccessTokens(projectId);
+    this.subscriptions$.push(personalAccessTokens$.subscribe((personalAccessTokens) => { this.personalAccessTokens = personalAccessTokens.map((token) => { return { ...token, scope: this.convertTokenScope(token.scope), expiresAt: this.convertTokenDate(token.expiresAt), createdAt: this.convertTokenDate(token.createdAt), lastUsed: this.convertTokenDate(token.lastUsed) } }) }));
+    return personalAccessTokens$;
+  }
+
+  handleWebsocketNotification(msgParts) {
+    if (msgParts[1] == 'pat') {
+      let id = msgParts[2];
+      let access_token_ids = this.personalAccessTokens.map((token) => token.id);
+      if (access_token_ids.includes(id)) {
+        this.personalAccessTokensQuery$.refetch();
+      }
+    }
+  }
+
+  deletePersonalAccessToken(tokenId: string) {
+    this.projectApolloService
+      .deletePersonalAccessTokenById(this.project.id, tokenId)
+      .pipe(first()).subscribe();
+  }
+
+  createPersonalAccessToken(name: string, expiresAt: string, scope: string) {
+    this.checkTokenNameIsDuplicated(name);
+    if (this.tokenNameIsDuplicated == false) this.projectApolloService.createPersonalAccessToken(this.project.id, name, expiresAt, scope).pipe(first()).subscribe((token) => this.newToken = token);
+  }
+
+  copyToken() {
+    this.tokenCopied = true;
+    copyToClipboard(this.newToken);
+    timer(1000).subscribe(() => this.tokenCopied = false);
+  }
+
+  closeTokenModal() {
+    this.newToken = null;
+    this.tokenNameIsDuplicated = null;
+  }
+
+  checkTokenNameIsDuplicated(name: string) {
+    for (const token of this.personalAccessTokens) {
+      if (token.name === name) {
+        this.tokenNameIsDuplicated = true;
+        return;
+      }
+    }
+    this.tokenNameIsDuplicated = false;
+  }
+
+  convertTokenScope(scope: string) {
+    if (scope == "READ") return "Read only";
+    else if (scope == "READ_WRITE") return "Read and write";
+    else return "Invalid";
+  }
+
+  convertTokenDate(date) {
+    if (date == null) return "Never";
+    return new Date(date).toDateString();
```

Comment:
might be wrong but i think we usually use a parse to for UTC & local return.

dateAsUTCDate(new Date(element.createdAt)).toLocaleString();